check.go

```diff
@@ -79,15 +79,20 @@ func (d *DBFT[H]) checkPreCommit() {
 		d.preBlockProcessed = true
 	}
 
-	// Require PreCommit sent by self for reliability. This condition may be removed
-	// in the future.
+	// Require PreCommit sent by self for reliability. This condition must not be
+	// removed because:
+	// 1) we need to filter out WatchOnly nodes;
```

Comment:
Technically they could assemble the block from messages and push it to clients even though they never sign anything. But that would be non-watch-only, so OK.

Reply:
> Technically they could assemble the block from messages and push it to clients

They do it. Once they got enough Commits:
```
2024-12-05T15:15:54.366+0300	DEBUG	dbft@v0.3.1/dbft.go:231	received message	{"type": "PreCommit", "from": 5, "height": 29, "view": 0, "my_height": 29, "my_view": 0}
2024-12-05T15:15:54.367+0300	DEBUG	dbft@v0.3.1/dbft.go:231	received message	{"type": "Commit", "from": 1, "height": 29, "view": 0, "my_height": 29, "my_view": 0}
2024-12-05T15:15:54.369+0300	INFO	dbft@v0.3.1/dbft.go:595	received Commit	{"validator": 1}
2024-12-05T15:15:54.369+0300	INFO	dbft@v0.3.1/check.go:118	approving block	{"height": 29, "hash": "0x8ed2b2803f34840c7d1289fba830468164a777ec941c253d2212690e932fed68", "tx_count": 0, "merkle": "0x3b5be8f1bd2248b0f39a02dd4d5ef110bcd85f3ee3d9ee41687caa0410533a4a", "prev": "0x53e640d375320e98bca6eb6ff01939bae23dce925612a01779285e10959a7ce9"}
```
```